components/api_request_helper/api_request_helper.cc

```diff
@@ -417,24 +453,36 @@ void APIRequestHelper::URLLoaderHandler::ParseSSE(
   static constexpr char kDataPrefix[] = "data: {";
 
   for (const auto& data : stream_data) {
+    DVLOG(3) << "Received chunk: " << data;
     if (!base::StartsWith(data, kDataPrefix)) {
+      VLOG(1) << "Data did not start with SSE prefix";
       continue;
     }
 
     std::string json = data.substr(strlen(kDataPrefix) - 1);
-
     auto on_json_parsed =
         [](base::WeakPtr<APIRequestHelper::URLLoaderHandler> handler,
            data_decoder::DataDecoder::ValueOrError result) {
+          VLOG(1) << "on_json_parsed";
           if (!handler) {
             return;
           }
+          handler->current_decoding_operation_count_--;
           DCHECK(handler->data_received_callback_);
+          VLOG(1) << "calling received";
           handler->data_received_callback_.Run(std::move(result));
+          if (!handler->data_chunk_parsed_callback_.is_null()) {
+            handler->data_chunk_parsed_callback_.Run();
+          }
         };
-
-    api_request_helper_->data_decoder_->ParseJson(
-        json, base::BindOnce(on_json_parsed, weak_ptr_factory_.GetWeakPtr()));
+    // Keep track of number of in-progress data decoding operations
+    // so that we can know if any are still in-progress when the request
+    // completes.
+    VLOG(1) << "Going to call ParseJson";
+    current_decoding_operation_count_++;
+    GetDataDecoder()->ParseJson(json,
```

Comment:
perf: `std::move(json)`

Reply:
changed, but also changed to `StringPiece`